File: src/csharp/Intel/Iced/Intel/NasmFormatterInternal/Registers.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if NASM
using Iced.Intel.FormatterInternal;

namespace Iced.Intel.NasmFormatterInternal {
	static class Registers {
		public static readonly FormatterString[] AllRegisters = GetRegisters();
		static FormatterString[] GetRegisters() {
			var registers = RegistersTable.GetRegisters();
			for (int i = 0; i < 8; i++)
				registers[(int)Register.ST0 + i] = new FormatterString("st" + i.ToString());
			return registers;
		}
	}
}
#endif

File: src/java/iced-x86/src/main/java/com/github/icedland/iced/x86/fmt/nasm/Registers.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.fmt.nasm;

import com.github.icedland.iced.x86.Register;
import com.github.icedland.iced.x86.internal.fmt.FormatterString;
import com.github.icedland.iced.x86.internal.fmt.RegistersTable;

final class Registers {
	public static final FormatterString[] allRegisters = getRegisters();

	static FormatterString[] getRegisters() {
		FormatterString[] registers = RegistersTable.getRegisters();
		for (int i = 0; i < 8; i++)
			registers[Register.ST0 + i] = new FormatterString("st" + i);
		return registers;
	}
}
